node/node.go

```diff
@@ -323,7 +324,22 @@ func doHandshake(
 	proxyApp proxy.AppConns,
 	consensusLogger log.Logger,
 ) error {
-	handshaker := cs.NewHandshaker(stateStore, state, blockStore, genDoc)
+	var err error
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+	stateProvider, err := statesync.NewLightClientStateProvider(
```

Comment:
If we are not using this "local sync", this would error, right?

We have to still support the case in which the node is simply recovering, and will restore all state from disk.